src/js/events/MetronomeClient.ts

```diff
@@ -2,33 +2,107 @@ import { request } from "@dcos/http-service";
 // TODO: remove this disable with https://jira.mesosphere.com/browse/DCOS_OSS-3579
 // tslint:disable-next-line:no-submodule-imports
 import { Observable } from "rxjs/Observable";
-
 import Config from "../config/Config";
 
 // Add interface information: https://jira.mesosphere.com/browse/DCOS-37725
-interface IJobData {
+export interface JobResponse {
+  id: string;
+  labels?: object;
+  run: {
+    cpus: number;
+    mem: number;
+    disk: number;
+    cmd: string;
+    env: object;
+    placement: {
+      constraints: any[];
+    };
+    artifacts: any[];
+  };
+  schedules: Schedule[];
+  historySummary: {
+    failureCount: number;
+    lastFailureAt: string | null;
+    lastSuccessAt: string | null;
+    successCount: number;
+  };
+}
+
+export interface JobDetailResponse {
   id: string;
+  description: string;
+  labels: object;
+  run: {
+    cpus: number;
+    mem: number;
+    disk: number;
+    cmd: string;
+    env: {};
+    placement: {
+      constraints: any[];
+    };
+    artifacts: any[];
+    maxLaunchDelay: number;
+    volumes: any[];
+    restart: {
+      policy: string;
+    };
+    docker?: {
+      secrets: object;
+      forcePullImage: boolean;
+      image: string;
+    };
+    secrets: object;
+  };
+  schedules: any[];
+  activeRuns: any[];
+  history: JobHistory;
 }
-interface IJobResponse {
+export interface Schedule {
+  concurrencyPolicy: string;
+  cron: string;
+  enabled: boolean;
   id: string;
+  nextRunAt: string;
+  startingDeadlineSeconds: number;
+  timezone: string;
 }
-interface IScheduleData {
+
+export interface JobHistory {
+  successCount: number;
+  failureCount: number;
+  lastSuccessAt: string;
+  lastFailureAt: null;
+  successfulFinishedRuns: JobHistoryRun[];
+  failedFinishedRuns: JobHistoryRun[];
+}
+
+export interface JobHistoryRun {
   id: string;
+  createdAt: string;
+  finishedAt: string;
 }
 
+export interface JobData {
+  id: string;
+}
+
+export interface ScheduleData {
+  id: string;
+}
 const defaultHeaders = {
   "Content-Type": "application/json; charset=utf-8"
 };
 
-export function createJob(data: IJobData) {
+export function createJob(data: JobData) {
```

Comment:
shall we specify return type?

Reply:
Yes, whenever we use these functions I would say (https://github.com/dcos/dcos-ui/pull/3006#discussion_r194408693)